src/phongo_bson_encode.c

```diff
@@ -466,6 +466,15 @@ static void php_phongo_zval_to_bson_internal(zval* data, php_phongo_field_path*
 			}
 
 			if (instanceof_function(Z_OBJCE_P(data), php_phongo_packedarray_ce)) {
+				/* If we are at the root-level, PackedArray instances should be
+				 * prohibited unless PHONGO_BSON_ALLOW_ROOT_ARRAY is set. */
+				bool is_root_level = (field_path->size == 0);
```

Comment:
I broke this out into its own variable for readability, since `(field_path->size == 0)` might be intuitive. Not concerned with overhead here, as it seems like something a compiler should easily optimize out.

Note: I didn't bother to share this variable with the enum and Type checks that follow later in the `switch` statement, since the comment there still applies:

> For the error handling that follows, we can safely assume that we are at the root level, since php_phongo_bson_append_object would have already been called for a non-root level.